### ISSUE TITLE
Version 3.8.0 shuts down after searching for an entry

### ISSUE THREAD

**Issue by vivadavid** [kiwix/kiwix-android#3556]
**Describe the bug**
When I search for an entry, the app shuts down.

**Steps to reproduce the behavior:**
1. I click on the Spanish Wikipedia Maxi or on the Spanish Wiktionary Maxi (I presume the issue applies to other ZIM files), which I have previously sideloaded.
2. I type something on the search box.
3. I click on the result I'm interested in.
4. The app shuts down (sometimes you see the actual entry for a few seconds, before the app shuts down).

**Environment**
- Version of Kiwix Android: 3.8.0. I was a beta tester on Google Play, but after waiting for some time to get 3.8.0, I signed out, and since being on the stable branch didn't make me get 3.8.0 either, I was impacient and, after uninstalling the app, I manually installed the universal APK.
- Device: Xiaomi mi 9T Pro.
- OS version:  Android 11.

**Comment by ManzoniGiuseppe** [comment]
I have something similar. In my csse 3.8.0 upgraded from 3.7.1 both from apk. In my case I found that if I write a letter at a time and then wait for the results, the search works, but if I write normally it crashes when it displays the results or immediately afterward

**Comment by kelson42** [comment]
@MohitMaliFtechiz This is a core feature regression here. Please fix in top priority.

**Comment by MohitMaliFtechiz** [comment]
@kelson42 On it.

**Comment by MohitMaliFtechiz** [comment]
@vivadavid, @ManzoniGiuseppe can you please try with this APK https://drive.google.com/file/d/190LJdxa9At4j3RmnonO4ibUMZB0Cup9L/view?usp=sharing , and let me know if this issue is still exist on your end.

**Comment by vivadavid** [comment]
> @vivadavid, @ManzoniGiuseppe can you please try with this APK https://drive.google.com/file/d/190LJdxa9At4j3RmnonO4ibUMZB0Cup9L/view?usp=sharing , and let me know if this issue is still exist on your end.

I've just tested it and it works: thanks!

Can I keep using it until a new official APK is released or is it safer to go back to 3.7.1?

**Comment by MohitMaliFtechiz** [comment]
> I've just tested it and it works: thanks!


> @MohitMaliFtechiz It works also for me, thank you.



Great. Thanks for testing, your feedback is valuable for us.

> Can I keep using it until a new official APK is released or is it safer to go back to 3.7.1?

For now, you can use this APK, but I suggest after merging https://github.com/kiwix/kiwix-android/pull/3558 please install latest nightly APK. from https://download.kiwix.org/nightly/.

**Comment by vivadavid** [comment]
> For now, you can use this APK, but I suggest after merging #3558 please install latest nightly APK. from https://download.kiwix.org/nightly/.

Ok, thanks!

**Comment by MohitMaliFtechiz** [comment]
@kelson42 The fix has been placed for this issue. 

> prepare everything for 3.8.1 please.

Okay.

**Comment by vivadavid** [comment]
Hi, this bug has been fixed in version 3.8.1 (thanks about that), but in this release, there's no APK.

Will it be published in the next few days?

**Comment by vivadavid** [comment]
I've just read there was a problem with the build. I'll keep visiting the Releases section.

**Comment by vivadavid** [comment]
> 3.8.1 has been release and is in review for beta by playstore team

Thanks for the information!

**Comment by kelson42** [comment]
> * I thought the APK would always available before the Google Play version and that it was even recommended. I planned to use the APK from now on, but I can stick to Google Play if it's recommended.

https://download.kiwix.org/release/kiwix-android/kiwix-3.8.1.apk
 
> * Version 3.8.1 can't find my side-loaded ZIM files. My previous version, the one I got here from Google Drive on 28 November, was fine. The thing is that version 3.8.1 doesn't ask for permission to access your files and I thought the problem was there, but after clearing the cache and allowing permission myself, it still can't find the ZIM files.

Look please at the [README](https://github.com/kiwix/kiwix-android/blob/develop/README.md) introduction

**Comment by ManzoniGiuseppe** [comment]
I don't know if I should open a new issue, but I switched from the apk in the link above to the official apk 3.8.1 and from time to time the app still crashes. The way I managed to reproduce it somewhat reliably is the following: write something e.g "super" or "paladin" and cancel the letters one by one, timing the cancel with the update of the results (all the updates take roughly the same), usually the app crashes before I run out of letters. Maybe I'm unlucky and I make too many mistakes, because it crashed 3 times with normal usage. Or maybe it happens also on other occasions and I found this one.
